elliottlib/cli/find_builds_cli.py

```diff
@@ -307,49 +289,129 @@ def _find_shipped_builds(build_ids: List[Union[str, int]], brew_session: koji.Cl
+    not_attachable_nvrs: Set[str] = set()
+
+    def add_builds(component_builds):
+        for component, build in component_builds.items():
+            if component not in attachable_components:
+                not_attachable_nvrs.add(build["nvr"])
+            elif build["id"] not in qualified_builds:
+                qualified_builds[build["id"]] = build
+
+    for tag in tag_pv_map:
+        # keys are rpm component names, values are nvres
+        component_builds: Dict[str, Dict] = builder.from_tag("rpm", tag, inherit=True, assembly=assembly, event=brew_event)
+        attachable_components |= component_builds.keys()  # components from our tag are always attachable
+
+        if runtime.assembly_basis_event:
+            # If an assembly has a basis event, rpms pinned by "is" and group dependencies should take precedence over every build from the tag
+            el_version = isolate_el_version_in_brew_tag(tag)
+            if not el_version:
+                continue   # Only honor pinned rpms if this tag is relevant to a RHEL version
+
+            # Honors pinned NVRs by "is"
+            pinned_by_is = builder.from_pinned_by_is(el_version, runtime.assembly, runtime.get_releases_config(), runtime.rpm_map)
+
+            # Builds pinned by "is" should take precedence over every build from tag
+            for component, pinned_build in pinned_by_is.items():
+                if component in component_builds and pinned_build["id"] != component_builds[component]["id"]:
+                    LOGGER.warning("Swapping stream nvr %s for pinned nvr %s...", component_builds[component]["nvr"], pinned_build["nvr"])
+            attachable_components |= pinned_by_is.keys()  # ART-managed rpms are always attachable
+            component_builds.update(pinned_by_is)  # pinned rpms take precedence over those from tags
+
+            # Honors group dependencies
+            group_deps = builder.from_group_deps(el_version, runtime.group_config, runtime.rpm_map)  # the return value doesn't include any ART managed rpms
+            # Group dependencies should take precedence over anything previously determined except those pinned by "is".
+            for component, dep_build in group_deps.items():
+                if component in component_builds and dep_build["id"] != component_builds[component]["id"]:
+                    LOGGER.warning("Swapping stream nvr %s for group dependency nvr %s...", component_builds[component]["nvr"], dep_build["nvr"])
+            component_builds.update(group_deps)
+
+        add_builds(component_builds)
+
+    # Will add all image member dependencies and RHCOS dependencies to the list of qualified builds we have found.
+    # Note if multiple unshipped NVRs for the same component and product version are present, Errata will probably only keep only 1 NVR.
+
+    all_el_versions = {isolate_el_version_in_brew_tag(tag) for tag in tag_pv_map}
+
+    # Honor image member dependencies
```

Comment:
Sorry, I failed to communicate this.. There will be `standard` and `custom` assemblies. Only `standard` assemblies can be attached to errata. It is invalid for a standard assembly to have either image specific dependencies or `rhcos` specific dependencies. The reason is that we need standard releases to have consistent RPMs through ART images & rhcos. The only way for that to happen is if `dependencies` are defined at the group.dependency level.
 
These other dependency fields (image member & rhcos) are only for `custom` assemblies where we have all sorts of flexibility in what we deliver to customers for a hotfix (i.e. we are not constrained by the normal rules of the errata workflow).

If you find member specific or rhcos specific dependencies defined, you can throw an exception.

Reply:
Thanks. I will fix that in the next push.